src/main/python/fine_tuning/run_batch.py

```diff
@@ -0,0 +1,211 @@
+            '-index', index_path,
+            '-topics', os.path.join(collection_yaml['anserini_root'], collection_yaml['topic_root'], collection_yaml['topic']),
+            '-{}'.format(models_yaml['basemodel']),
+            para[0],
+            '-output', para[1]
+        )
+        all_params.append(this_para)
+    logger.info('='*10+'Starting Batch Retrieval'+'='*10)
+    batch_everything(all_params, atom_retrieval)
+
+def atom_retrieval(para):
+    subprocess.call(' '.join(para), shell=True)
+
+def batch_eval(collection_yaml, models_yaml, output_root):
+    all_params = []
+    index_path = get_index_path(collection_yaml)
+    programs = set([eval['command']+' '+eval['params'] for eval in collection_yaml['evals']])
+    this_output_root = os.path.join(output_root, collection_yaml['name'])
+    if not os.path.exists(this_output_root):
+        os.makedirs(this_output_root)
+    eval_params = Evaluation(index_path).gen_batch_eval_params(this_output_root)
+    for para in eval_params:
+        run_file_path, eval_output = para
+        this_para = (
+            [os.path.join(collection_yaml['anserini_root'], program) for program in programs],
+            os.path.join(collection_yaml['anserini_root'], collection_yaml['qrels_root'], collection_yaml['qrel']),
+            run_file_path,
+            eval_output
+        )
+        all_params.append(this_para)
+    logger.info('='*10+'Starting Batch Evaluation'+'='*10)
+    batch_everything(all_params, atom_eval)
+
+def atom_eval(params):
+    Evaluation.output_all_evaluations(*params)
+
+def batch_output_effectiveness(collection_yaml, models_yaml, output_root):
+    all_params = []
+    index_path = get_index_path(collection_yaml)
+    this_output_root = os.path.join(output_root, collection_yaml['name'])
+    if not os.path.exists(this_output_root):
+        os.makedirs(this_output_root)
+    all_params.extend( Effectiveness(index_path).gen_output_effectiveness_params(this_output_root) )
+    logger.info('='*10+'Starting Output Effectiveness'+'='*10)
+    batch_everything(all_params, atom_output_effectiveness)
+
+def atom_output_effectiveness(para):
+    index_path = para[0]
+    output_fn = para[1]
+    input_fns = para[2:]
+    Effectiveness(index_path).output_effectiveness(output_fn, input_fns)
+
+def print_optimal_effectiveness(collection_yaml, models_yaml, output_root, metrics=['map']):
+    index_path = get_index_path(collection_yaml)
+    this_output_root = os.path.join(output_root, collection_yaml['name'])
+    logger.info('='*30+'Optimal Effectiveness for '+collection_yaml['name']+'='*30)
+    effectiveness = Effectiveness(index_path).load_optimal_effectiveness(this_output_root, metrics)
+    print(json.dumps(effectiveness, sort_keys=True, indent=2))
+    return
+    success = True
+    for e in effectiveness:
+        expected = models_yaml[e['model']]['expected'][collection_yaml['name']][e['metric']][e['topic']]
+        if isclose(expected, e['actual']):
```

Comment:
There is not key named `actual`. The valid keys are `actual_optimal` and `actual_all`

Reply:
yea, I will remove the code after the `return`